typescript/cli/src/utils/warp.ts

```
@@ -33,3 +50,147 @@ export async function getWarpCoreConfigOrExit({
+  assert(
+    warpDeployConfig,
+    `Missing warp deploy config for warp route ${selectedId}.`,
+  );
+
+  const filledConfig = await fillDefaults(context, warpDeployConfig);
+  const validatedConfig =
+    WarpRouteDeployConfigMailboxRequiredSchema.parse(filledConfig);
+
+  return {
+    warpDeployConfig: validatedConfig,
+    warpCoreConfig,
+  };
+}
+
+/**
+ * Compares chains between warp deploy and core configs, filters them to only include matching chains,
+ * and logs warnings if there are mismatches.
+ * @param warpDeployConfig The warp deployment configuration
+ * @param warpCoreConfig The warp core configuration
+ * @returns The filtered warp deploy and core configs containing only matching chains
+ */
+export function filterWarpConfigsToMatchingChains(
+  warpDeployConfig: WarpRouteDeployConfigMailboxRequired,
+  warpCoreConfig: WarpCoreConfig,
+): {
+  warpDeployConfig: WarpRouteDeployConfigMailboxRequired;
+  warpCoreConfig: WarpCoreConfig;
+} {
+  const deployConfigChains = Object.keys(warpDeployConfig);
+  const coreConfigChains = warpCoreConfig.tokens.map(
+    (t: { chainName: string }) => t.chainName,
+  );
+
+  const deploySet = new Set(deployConfigChains);
+  const coreSet = new Set(coreConfigChains);
+
+  if (!setEquality(deploySet, coreSet)) {
+    logRed(
+      'Warning: Chain mismatch between warp core config and warp deploy config:',
+    );
+    logRed('──────────────────────');
+    logRed('Deploy config chains:');
+    deployConfigChains.forEach((chain: string) => logRed(`  - ${chain}`));
+    logRed('Core config chains:');
+    coreConfigChains.forEach((chain: string) => logRed(`  - ${chain}`));
+
+    const matchingChains = intersection(deploySet, coreSet);
+    if (matchingChains.size === 0) {
+      logRed('Error: No matching chains found between configs');
+      process.exit(1);
+    }
+
+    logRed(
+      `Continuing with check for matching chains: ${Array.from(
+        matchingChains,
+      ).join(', ')}\n`,
+    );
```

Just a small nit on my side, but I would prefer the command to exit if there are mismatches between the input files, as only reading the matching chains might silently hide user errors that might end up in a wrong check result too